kernel/kernel-api/src/test/scala/io/delta/kernel/internal/checksum/ChecksumWriterSuite.scala

```diff
@@ -0,0 +1,191 @@
+    checkMetadata(metadata, row.getStruct(METADATA))
+    checkProtocol(protocol, row.getStruct(PROTOCOL))
+  }
+
+  private def createTestMetadata(): Metadata = {
+    new Metadata(
+      "id",
+      Optional.of("name"),
+      Optional.of("description"),
+      new Format("parquet", Collections.emptyMap()),
+      "schemaString",
+      new StructType(),
+      stringArrayValue(util.Arrays.asList("c3")),
+      Optional.of(123),
+      stringStringMapValue(new util.HashMap[String, String]() {
+        put("delta.appendOnly", "true")
+      })
+    )
+  }
+
+  private def createTestProtocol(): Protocol = {
+    new Protocol(
+      /* minReaderVersion= */ 1,
+      /* minWriterVersion= */ 2,
+      Collections.emptyList(),
+      Collections.emptyList()
+    )
+  }
+
+  private def checkMetadata(metadata: Metadata, metadataRow: Row): Unit = {
+    assert(metadataRow.getSchema == Metadata.FULL_SCHEMA)
+
+    def getOptionalString(field: String): Optional[String] =
+      Optional.ofNullable(metadataRow.getString(Metadata.FULL_SCHEMA.indexOf(field)))
+
+    assert(metadataRow.getString(Metadata.FULL_SCHEMA.indexOf("id")) == metadata.getId)
+    assert(getOptionalString("name") == metadata.getName)
+    assert(getOptionalString("description") == metadata.getDescription)
+
+    val formatRow = metadataRow.getStruct(Metadata.FULL_SCHEMA.indexOf("format"))
+    assert(
+      formatRow.getString(Format.FULL_SCHEMA.indexOf("provider")) == metadata.getFormat.getProvider
+    )
+
+    assert(
+      metadataRow
+        .getString(Metadata.FULL_SCHEMA.indexOf("schemaString")) == metadata.getSchemaString
+    )
+    assert(
+      metadataRow
+        .getArray(Metadata.FULL_SCHEMA.indexOf("partitionColumns")) == metadata.getPartitionColumns
+    )
+    assert(
+      Optional
+        .ofNullable(metadataRow.getLong(Metadata.FULL_SCHEMA.indexOf("createdTime")))
+      == metadata.getCreatedTime
+    )
+    assert(
+      VectorUtils
+        .toJavaMap(metadataRow.getMap(Metadata.FULL_SCHEMA.indexOf("configuration")))
+      == metadata.getConfiguration
+    )
+  }
+
+  private def checkProtocol(protocol: Protocol, protocolRow: Row): Unit = {
+    assert(protocolRow.getSchema == Protocol.FULL_SCHEMA)
+    assert(
+      protocol.getMinReaderVersion == protocolRow
+        .getInt(Protocol.FULL_SCHEMA.indexOf("minReaderVersion"))
+    )
+    assert(
+      protocol.getMinWriterVersion == protocolRow
+        .getInt(Protocol.FULL_SCHEMA.indexOf("minWriterVersion"))
+    )
+  }
+}
+
+/**
+ * Mock implementation of JsonHandler for testing checksum file writing.
+ */
+class MockCheckSumFileJsonWriter extends BaseMockJsonHandler {
+  var capturedCrcRow: Row = new GenericRow(new StructType(), new util.HashMap[Integer, AnyRef])
+  var checksumFilePath: String = ""
+
+  override def writeJsonFileAtomically(
+    filePath: String,
+    data: CloseableIterator[Row],
+    overwrite: Boolean): Unit = {
```

Comment:
nit: should be 4 space indentation on continuation lines

Reply:
I manually fix it :-(